src/chat/zulip.md

```diff
@@ -0,0 +1,180 @@
+really helpful.
+
+You'll definitely want to configure the streams that you're subscribed to when
+getting started; the default set is quite limited, and there are many groups
+that exist beyond it. Subscribing to a stream is very low cost -- it is similar
+to being "in" an IRC channel, except that logs are available for all streams,
+regardless of subscription status.
+
+It's not necessary to introduce yourself, but feel free to say hello in the
+`#new members` stream.
+
+## User groups
+
+User groups can be pinged by anyone with the `@<group>` notation, same as
+pinging another user. Groups can be created by anyone, and anyone can join a
+group.
+
+Users should feel free to join (or leave) groups on their own. Furthermore,
+users should feel free to create groups as needed, though it is currently
+expected that this is somewhat rare. You should name your group similar to how
+you would name a stream for the same purpose, though groups can be more
+fine-grained (or less). For example, `@T-compiler/meeting` currently does not
+have a dedicated stream.
+
+## Appropriate conversation
+
+In most streams, you should try to keep conversations related to team business.
+The `#general` stream is a bit broader, but even there, discussions should be
+closely related to Rust (though may not relate to projects of any particular
+team). All channels are expected to be used for discussions related to the Rust
+project, though; discussions of (for example) wildlife or sightseeing are not
+appropriate.
+
+## Streams
+
+These are similar to "channels" on other platforms (i.e., there should not be
+too many). On the other hand, you can choose which streams you subscribe to, so
+there can be more than channels on other platforms. Read [Zulip's
+documentation](https://zulipchat.com/help/about-streams-and-topics) for more
+details.
+
+Streams are appropriate for any Rust official group. For example, working
+groups, project groups, teams are all examples of official groups. These should
+ideally also be represented in the [team repository](../infra/team-maintenance.md).
+
+### Default streams
+
+The default set of streams is chosen to allow incoming people to be able to have
+at least one place to go that can then, if necessary, direct them to a more
+specific location.
+
+Currently that means that every top-level group present on Zulip is by default
+visible. Specifically, no stream that contains a `/` will be enabled by default.
+
+Currently this set is:
+ * general
+ * t-lang
+ * t-compiler
+ * project-ffi-unwind
+ * project-inline-asm
+ * project-safe-transmute
+ * rust-survey-2019
+ * wg-async-foundations
+ * wg-database
+ * wg-formal-methods
+ * wg-secure-code
+ * wg-traits
+ * zulip
+
+FIXME: this may be too many -- in particular, I'm not sure how I'd justify this,
+but maybe a better default set is (#general, #zulip, #announce, #new members),
+with the expected getting started workflow for everyone being to setup
+subscribed streams.
```

Comment:
Hmm, I think these are the two most reasonable choices -- the nice thing about the more minimal set, I guess, is that it is sort of *obviously* too small. The larger set though definitely includes things most folks won't want (e.g., `rust-survey-2019`). This seems like one of the key things to resolve! 

I could see just including "t-*" and "wg-*" by default.

I also sort of think a channel like `#t-compiler/help` might be good to have in the default set. Hmm.

Reply:
Also, really rust-survey-2019 should probably be like t-core/rust-survey-2019 or t-community I guess

Reply:
One of the benefits of the minimal set, I think, is that it helps guide people right off the bat into more targeted channels, and allows us to have more of them. If we make (for example) only top-level streams prefixed by t/wg basically, I worry that it would be not too effective to add (for example) `#t-compiler/bootstrap` or something along those lines, as people would be hard pressed to find it.

Reply:
Yeah. The downside is that it's going to be more confusing when you arrive and are like "where is everything?"

Reply:
Indeed. I think it'd be interesting to try and talk to the Zulip devs about this -- they probably have more experience than us with the onboarding strategy that works best :)

There doesn't seem to be much guidance at https://zulipchat.com/help/set-default-streams-for-new-users. I guess maybe we can arbitrarily ping someone? Not sure how best to reach out :)

Maybe an email makes sense.

Reply:
I've updated the other comments and emailed Zulip support for this question to hopefully get some feedback from them. Let's see what comes out of that :)